### PR TITLE
add tests for checking synchronize switch action on activation

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        verifyAll(northbound.validateSwitch(sw.dpId)) {
+            it.meters.proper*.meterId.sort() == createdMeterIds.sort()
+            switchHelper.verifyMeterSectionsAreEmpty(it, ["excess", "missing", "misconfigured"])
+        }
+
+        and: "Cleanup: Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+    }
+
+    def "Excess rules/meters are synced from a new switch connected to the controller"() {
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy]: I'd say '...upon connecting to the controller' or '...before connecting..' or '..when connecting..'

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+class SwitchActivationSpec extends HealthCheckSpecification {
+    @Value("#{kafkaTopicsConfig.getSpeakerFlowTopic()}")
+    String flowTopic
+    @Autowired
+    @Qualifier("kafkaProducerProperties")
+    Properties producerProps
+    @Autowired
+    SwitchHelper switchHelper
+
+    def "Missing flow rules are synced from a new switch connected to the controller"() {
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy]: I agree when you say '_Excess_ rules are synced _FROM_...', but _missing_ rules are synced _to_ the switch. Or use 'removed'/'installed' verbs instead.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy]: changed to `Excess rules/meters are synced from a new switch before connecting`

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy]: fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy [hunk 3]
```diff
@@ -58,8 +57,8 @@ class FlowRulesSpec extends HealthCheckSpecification {
     }
 
     @Tags([VIRTUAL, SMOKE])
-    def "Pre-installed rules are not deleted from a new switch connected to the controller"() {
-        given: "A switch with some rules installed (including default) and not connected to the controller"
+    def "Pre-installed flow rules are not deleted from a new switch connected to the controller"() {
```
[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: > This test used to check that 'excess' rules are not getting removed on activation. 
from now on this test checks that proper rules are not deleted on activation.

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        verifyAll(northbound.validateSwitch(switchPair.src.dpId)) {
+            it.rules.proper.containsAll(createdCookies)
+            switchHelper.verifyRuleSectionsAreEmpty(it, ["missing", "excess"])
+        }
+
+        and: "Cleanup: Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+    }
+
+    def "Missing flow meters are synced from a new switch connected to the controller"() {
```
[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy]: that makes sense

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy (hunk 3 above)
[the remark this answers is not in the document]
Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: I understand. Did you check that there is no other test that already verifies this behavior?

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy (hunk 1 above)
Comment (rtretyak) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy]: I don't see the change

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+class SwitchActivationSpec extends HealthCheckSpecification {
+    @Value("#{kafkaTopicsConfig.getSpeakerFlowTopic()}")
+    String flowTopic
+    @Autowired
+    @Qualifier("kafkaProducerProperties")
+    Properties producerProps
+    @Autowired
+    SwitchHelper switchHelper
+
+    def "Missing flow rules/meters are removed from a new switch connected to the controller"() {
```
Comment (rtretyak) [inline services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy]: Missing rules are removed from the switch? Missing rules should be installed to the switch. Also, please rephrase '..new switch connected to the controller' to `..new switch upon connecting to the conntroller/when connecting to the controller/before connecting to the controller`

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy]: I don't have idea how it happened
sorry for any inconvenience
now it is totally fixed

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy (hunk 5 above)
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy]: fixed
`Missing flow rules/meters are installed on a new switch before connecting to the controller`

File: services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy (hunk 3 above)
[the remark this answers is not in the document]
Comment (andriidovhan) [reply services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/FlowRulesSpec.groovy]: I didn't find separate test for that purpose

**Pull request merged**